09-redux/classworks/Volodkova_Anna/src/App.js

```diff
@@ -0,0 +1,77 @@
+import React, {useEffect, useState, useRef} from 'react';
+import {useRouteMatch} from "react-router-dom";
+import {useSelector, useDispatch} from 'react-redux';
+import Header from 'components/header'
+import Pages from 'pages';
+import {withRedditApi} from './hooks/useRedditApi';
+import {setPostsNull, setSearchValue, fetchPosts} from "./resourses/actions";
+
+function App(props) {
+  const dispatch = useDispatch();
+  const getSearchValue = useSelector((state) => state.value);
+  let getSubreddits = useSelector((state)=>state.posts);
+  console.log('getSubreddits from app.js', getSubreddits);
+
+  const [reactSubreddit, setReactSubreddit] = useState(null);
+  const [reactAbout, setReactAbout] = useState(null);
+
+
+  const getdata = async (name = 'r/react') => {
+    const {fetchReddit} = props;
+    const data = await fetchReddit(`/${name}/hot`).then(res => res.json());
+    setReactSubreddit(data);
+    const about = await fetchReddit(`/${name}/about`).then(res => res.json());
+    setReactAbout(about);
+  };
+
+
+  useEffect(() => {
+    if (props.isToken) {
+      getdata();
+      dispatch(setPostsNull());
```

Comment:
you should not have such action like `setPostsNull`, or `setPostsAnotherVeryCoolValue` or `setPosts25`.
only `setPosts`, which gets value as a param